File: GO/NumberOfZeroFilledSubarrays.go
/**
 * @version GO 1.20.2
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

func zeroFilledSubarray(nums []int) int64 {
	var ans int64
	var c int

	for i := range nums {
		if nums[i] == 0 {
			c++
			ans += int64(c)
		} else {
			c = 0
		}
	}

	return ans
}

File: JS/NumberOfZeroFilledSubarrays.js
/**
 * @version JAVASCRIPT ECMAScript 6
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */

/**
 * @param {number[]} nums
 * @return {number}
 */
function zeroFilledSubarray(nums, val) {
    let zeros = 0, ans = 0;
    
    for (let num of nums) {
        zeros = num === 0 ? zeros + 1 : 0;
        ans += zeros;
    }
    
    return ans;
}
export { zeroFilledSubarray }
